### PR TITLE
Add hints to array destructuring error messages

### PR DESCRIPTION
**Pull request by MDLC01**
This small PR adds hints to array destructuring error messages indicating the length of the array. For example: "the provided array has a length of 3, but the pattern expects 2 elements".

### CODE REVIEW

File: crates/typst/src/eval/binding.rs [hunk 1]
```diff
@@ -89,23 +89,30 @@ fn destructure_array<F>(
 where
     F: Fn(&mut Vm, ast::Expr, Value) -> SourceResult<()>,
 {
+    let destruct_len = destruct.items().count();
```
Comment (laurmaedje) [inline crates/typst/src/eval/binding.rs:92]: why was this moved up here? it doesn't seem to be used anywhere but `Spread`, but now we unnecessarily traverse the destructuring AST twice.

File: crates/typst/src/eval/binding.rs [hunk 2]
```diff
@@ -89,23 +89,30 @@ fn destructure_array<F>(
...
     let mut i = 0;
 
     for p in destruct.items() {
         match p {
             ast::DestructuringItem::Pattern(pattern) => {
                 let Ok(v) = value.at(i as i64, None) else {
-                    bail!(pattern.span(), "not enough elements to destructure");
+                    bail!(
+                        pattern.span(), "not enough elements to destructure";
+                        hint: "the provided array has a length of {}", len
```
Comment (laurmaedje) [inline crates/typst/src/eval/binding.rs:102]:
```suggestion
                        hint: "the provided array has a length of {len}",
```

File: crates/typst/src/eval/binding.rs (hunk 1 above)
Comment (MDLC01) [reply crates/typst/src/eval/binding.rs:92]: This is indeed unnecessary.

**Comment by laurmaedje** [comment]
Thanks!
